### PR TITLE
 Count the number of subset with a given difference

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes

### Description

(Write your answer here.)
This problem is solved using dp.
In this problem we have to count whether two subset difference is equal to sum or not
if equal then we will return true else we will return false.
Input - 
4 1
1 1 2 3
Output - 3

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#5248

### CODE REVIEW

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -201,6 +201,7 @@ _add list here_
 - [Zero One Knapsack](dp/Zero_One_Knapsack.cpp)
 - [ZeroN Knapsack](dp/Zero_N_knapsack.cpp)
 - [Sprague Grundy Theorem on Strings](dp/Sprague_Grundy_Theorem.cpp)
+- [ Count the number of subset with a given difference](dp/Count_subset_given_difference.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove any extra spaces

File: C-Plus-Plus/dp/Count_subset_given_difference.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+This problem is solved using dp.
+In this problem we have to count whether two subset difference is equal to sum or not
+if equal then we will return true else we will return false.
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Count_subset_given_difference.cpp]: Give a detailed description for the problem

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/Count_subset_given_difference.cpp (hunk 2 above)
Comment (Meghag5) [reply C-Plus-Plus/dp/Count_subset_given_difference.cpp]: done

File: C-Plus-Plus/README.md (hunk 1 above)
Comment (Meghag5) [reply C-Plus-Plus/README.md]: done @HarshCasper pls review

File: C-Plus-Plus/dp/Count_subset_given_difference.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+This problem is solved using dp.
+In this we will have array and difference.
+We have to divide array in two subsets in such a way that difference of both subsets 
+will become equal to given difference.
+if equal then we will increase count.
+and if there is no such subset then return 0 else return count
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Count_subset_given_difference.cpp]: Use a more standard description

File: C-Plus-Plus/dp/Count_subset_given_difference.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            // if subset sum value less than sum then we will proceed further and we add value to it
+            else{
+                t[i][j] = t[i-1][j];
+            }
+        }
+    }
+    
+    // return total count
+    return t[n][sum];
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Count_subset_given_difference.cpp]: Format the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
done @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**